src/app/components/practice/question-authoring-page/question-authoring.guard.ts

```diff
@@ -16,8 +17,18 @@ export class QuestionAuthoringGuard implements CanActivate {
     next: ActivatedRouteSnapshot,
     state: RouterStateSnapshot): Observable<boolean> {
     return this.authenticationService.currentUser$.pipe(map((user) => {
-        return ((user.permissions.
-        has(UserPermission.ORG_ADMIN) || user.permissions.has(UserPermission.CREATOR)));
+        return (this.permissionCheck(user));
     }));
   }
+
+  permissionCheck(user: User){
+    if ((user.permissions.
+      has(UserPermission.ORG_ADMIN) || user.permissions.has(UserPermission.CREATOR))) {
+      return true;
+    } else {
+      window.alert('Apply for contributor permissions to view this page');
```

Comment:
I don't think this is the right means of solving the issue. I think we would want utilise some of the mechanisms that the clarity UI makes available for solving this problem

Reply:
ok, I kind of briefly looked around and didn't see much, maybe point me in the right direction, or if you wan to do this one I can revert the change?

Reply:
Sure. One idea I have in mind is that we could make use of something called a modal.

https://clarity.design/documentation/modals

An alert, https://clarity.design/documentation/alerts, might also be a good solution. 

I think a modal might be a better option because they lend themselves toward allowing the user to take the action of applying for the permission they are missing. We could put a little form in it (doesn't have to be hooked up properly an endpoint somewhere) that allows them this request on the model. If you fancy giving it a go, you're welcome 😊. If you'd rather I did, let me know as well.

Reply:
Doing some more research on common UI patterns, an alternative option might be to allow the user navigate to the page they want, but show them an error message, instead of the page content. For example, here's what GitHub shows me if I try to edit content on a public repository that I am not a member of:
![image](https://user-images.githubusercontent.com/20339399/94773200-364d5880-03b3-11eb-9c7b-68a92fa0dc37.png)

Reply:
this is now implemented in: #74